src/pds/peppi/query_builder.py

```diff
@@ -121,22 +122,54 @@ def _add_clause(self, clause, logical_join="and"):
         else:
             self._q_string = clause
 
-    def has_target(self, identifier: str):
-        """Adds a query clause selecting products having a given target identifier.
+    def _has_target(self, identifiers: Union[list, str]):
+        """Adds a query clause from 1 or n, target lids, apply OR operator between lids."""
+        if isinstance(identifiers, str):
+            identifiers = [identifiers]
+
+        if len(identifiers) > 0:
+            clause = "or ".join([f'ref_lid_target eq "{identifier}"' for identifier in identifiers])
```

Comment:
A leading space before the `or` here might be needed here as well?

Reply:
I think that works without but that will be cleaner with.